src/psyclone/psyGen.py

```diff
@@ -1502,15 +1503,19 @@ def sameParent(self, node_2):
             return True
         return False
 
-    def walk(self, my_type):
+    def walk(self, my_type, stop_type=None):
         ''' Recurse through the PSyIR tree and return all objects that are
         an instance of 'my_type', which is either a single class or a tuple
         of classes. In the latter case all nodes are returned that are
         instances of any classes in the tuple.
 
         :param my_type: the class(es) for which the instances are collected.
-        :type my_type: either a single :py:class:`psyclone.Node` class\
-            or a tuple of such classes.
+        :type my_type: either a single :py:class:`psyclone.Node` class \
+            or a tuple of such classes
+        :param stop_type: the class(es) that will not be recursed into \
```

Comment:
How about: "class(es) at which recursion is halted (optional)". Also, could you mention this parameter in the general description of the method in order to explain its purpose.

Reply:
Done